node/src/components/deploy_buffer.rs

```diff
@@ -92,27 +112,41 @@ pub(crate) struct DeployBuffer {
 
 impl DeployBuffer {
     /// Creates a new, empty deploy buffer instance.
-    pub(crate) fn new(block_max_deploy_count: usize) -> Self {
-        DeployBuffer {
+    pub(crate) fn new<REv>(
+        effect_builder: EffectBuilder<REv>,
+        block_max_deploy_count: usize,
+    ) -> (Self, Effects<Event>)
+    where
+        REv: ReactorEventT,
+    {
+        let this = DeployBuffer {
             block_max_deploy_count,
-            collected_deploys: HashMap::new(),
-            processed: HashMap::new(),
+            pending: HashMap::new(),
+            proposed: HashMap::new(),
             finalized: HashMap::new(),
             chainspecs: HashMap::new(),
-        }
+        };
+        let cleanup = effect_builder
+            .set_timeout(DEPLOY_BUFFER_PRUNE_INTERVAL)
+            .event(|_| Event::BufferPrune);
+        (this, cleanup)
     }
 
     /// Adds a deploy to the deploy buffer.
     ///
     /// Returns `false` if the deploy has been rejected.
-    fn add_deploy(&mut self, hash: DeployHash, header: DeployHeader) {
+    fn add_deploy(&mut self, current_instant: Timestamp, hash: DeployHash, header: DeployHeader) {
+        if header.expired(current_instant) {
+            info!("expired deploy {} rejected from the buffer", hash);
```

Comment:
Maybe `debug` or `trace` even? We don't want to spam logs on `info` level with events that are very cheap and easy to trigger.

Reply:
`trace!` it is!